### DISCUSSION TITLE
Sprite from atlas showing other part of atlas

### DISCUSSION THREAD

**Discussion by propah** [bevyengine/bevy#4424] [Q&A] [answered]
Hello, I just started a project using a sprite in a TextureAtlas. The problem is when I perform a transform on the sprite, it shows pixels that are in an other part of the atlas. This is strange because the texture is 32x32 and the little green lines aren't even a full pixel large. I'm not sure how the bevy engine works, maybe someone can explain me? 
![image](https://user-images.githubusercontent.com/79640420/161813800-dcfa3bff-cc0d-4dd4-8b84-c010c8037c08.png)

**Comment by NiklasEi** [comment, marked as answer]
One very common way of preventing this is to add padding to your sprite sheet. This can be done automatically in software like e.g. Aseprite. Bevy also supports reading sprite sheets with padding ([see from_grid_with_padding](https://docs.rs/bevy/latest/bevy/prelude/struct.TextureAtlas.html#method.from_grid_with_padding)).

**Comment by wilk10** [comment]
I think this is related to https://github.com/bevyengine/bevy/issues/4162 . I have run into the same issue. Padding the sprites is definitely a workaround, but it might affect users workflow when producing them. Hopefully Bevy can handle it internally
